src/compiler/types.ts

```diff
@@ -1674,8 +1674,8 @@ module ts {
         properties: Symbol[];              // Properties
         callSignatures: Signature[];       // Call signatures of type
         constructSignatures: Signature[];  // Construct signatures of type
-        stringIndexType: Type;             // String index type
-        numberIndexType: Type;             // Numeric index type
+        stringIndexType?: Type;            // String index type
+        numberIndexType?: Type;            // Numeric index type

```

Comment:
Possibly, they all get assigned currently so no need for null check:
https://github.com/Microsoft/TypeScript/blob/master/src/compiler/checker.ts#L1066